lib/auxiliary/sysinfo_parser.rb

```diff
@@ -0,0 +1,186 @@
+# frozen_string_literal: true
+
+require 'active_support'
+require 'active_support/core_ext/object/deep_dup'
+
+# AutoHCK module
+module AutoHCK
+  # UtilsParser class
+  class UtilsParser
+    # Converts string and float input to int. Strips all non-numeric
+    # characters from strings.
+    #
+    # @param value [String, Float] Input value
+    #
+    # @return [Integer, nil] Integer if successful conversion, otherwise nil
+    def self.convert_to_int(value)
+      if value.is_a?(String)
+        str_val = value.gsub(/[^0-9\-.]/, '')
+        begin
+          Integer(str_val)
+        rescue ArgumentError, TypeError
+          begin
+            Integer(Float(str_val))
+          rescue ArgumentError, TypeError
+            nil
+          end
+        end
+      elsif value.is_a?(Numeric)
+        value.to_i
+      end
+    end
+  end
+
+  # SysInfoParser class
+  class SysInfoParser
+    DELIM = ':' # k/v delimiter
+
+    # Main text parsing function
+    #
+    # @param data [String] Text data to parse
+    #
+    # @return [Array<Hash>] List of dictionaries. Processed structured data.
+    def parse(data)
+      lines = data.lines.map(&:chomp).reject(&:empty?)
+
+      # find the character position of the value in the k/v pair of the first line
+      # all subsequent lines of data use the same character position
+      start_value_pos = get_value_pos(lines[0], DELIM)
```

Comment:
`systeminfo` has an option to CSV so you should use it when parsing mechanically.

Reply:
CSV format is not fully helpful. Fields hotfixes, nic, processors, etc are still is a multiline data and some of them should be parsed.

Reply:
Yes, but I think it will still save several lines and interface stability.